packages/astro/src/runtime/server/astro-island.ts

```diff
@@ -129,7 +134,7 @@ declare const Astro: {
 					window.dispatchEvent(new CustomEvent('astro:hydrate'));
 				};
 				attributeChangedCallback() {
-					if (this.hydrator) this.hydrate();
+					this.hydrate();
```

Comment:
NOTE: `this.hydrate` already has the `this.hydrator` guard, so this can be simplified.